src/libgit2/config_cache.c

```diff
@@ -64,11 +64,10 @@ static git_configmap _configmap_logallrefupdates[] = {
 	{GIT_CONFIGMAP_STRING, "always", GIT_LOGALLREFUPDATES_ALWAYS},
 };
 
-/*
- * Generic map for integer values
- */
-static git_configmap _configmap_int[] = {
+static git_configmap _configmap_abbrev[] = {
+	{GIT_CONFIGMAP_FALSE, NULL, GIT_ABBREV_FALSE},
 	{GIT_CONFIGMAP_INT32, NULL, 0},
+	{GIT_CONFIGMAP_STRING, "auto", GIT_ABBREV_DEFAULT}
```

Comment:
I think that you need to invert the bool and int cases so that `0` can be correctly identified as an invalid input, but `false` is allowed. Otherwise, we treat the `0` as falsey.

```suggestion
static git_configmap _configmap_abbrev[] = {
	{GIT_CONFIGMAP_INT32, NULL, 0},
	{GIT_CONFIGMAP_FALSE, NULL, GIT_ABBREV_FALSE},
	{GIT_CONFIGMAP_STRING, "auto", GIT_ABBREV_DEFAULT}
```